packages/todos_api/test/src/todo_test.dart

```diff
@@ -0,0 +1,106 @@
+// ignore_for_file: prefer_const_constructors
+import 'package:test/test.dart';
+import 'package:todos_api/todos_api.dart';
+
+void main() {
+  group('Todo', () {
+    const itemId = '123';
+    const itemTitle = 'clean';
+    const itemDescription = 'little description';
+    const isCompleted = true;
+
+    test('have generated, not empty id if none is provided', () {
+      final todo = Todo(title: itemTitle);
+      expect(todo.id, isNotEmpty);
+    });
+
+    test('accepts passed id if one is provided', () {
+      final todo = Todo(id: itemId, title: itemTitle);
+      expect(todo.id, equals(itemId));
+    });
+
+    test('throws an error if empty id is provided', () {
+      const todoId = '';
+      expect(
+        () {
+          Todo(id: todoId, title: 'title');
+        },
+        throwsA(isA<AssertionError>()),
+      );
+    });
+
+    test('two objects with the same fields are equal', () {
+      final todoItem = Todo(id: itemId, title: itemTitle);
+      final todoItem2 = Todo(id: itemId, title: itemTitle);
+      expect(todoItem, equals(todoItem2));
+    });
```

Comment:
My preferred way to test for value equality would compare 3 objects: 2 that should be equal and 1 that's different. This way you check for positive and negative case.

```suggestion
    test('supports value equality', () {
      final todoItem = Todo(id: itemId, title: itemTitle);
      final todoItem2 = Todo(id: itemId, title: itemTitle);
      final todoItem3 = Todo(id: itemId, title: itemTitle, isCompleted: false);
      expect(todoItem, equals(todoItem2));
      expect(todoItem, isNot(equals(todoItem3)));
    });
```